lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -586,7 +637,36 @@ DiagnosticVerifier::Result DiagnosticVerifier::verifyFile(unsigned BufferID) {
       llvm::SMFixIt fix(llvm::SMRange(replStartLoc, replEndLoc), actual);
       addError(replStartLoc.getPointer(), "expected no fix-its; actual fix-it seen: " + actual, fix);
     }
-    
+
+    if (auto expectedNotes = expected.EducationalNotes) {
+      // Verify educational notes
+      for (auto &foundName : FoundDiagnostic.EducationalNotes) {
```

Comment:
Not sure how much this being quasi-quadratic in practice actually matters, but I thought I'd note it now.

Reply:
It'll be extremely rare to ever have more than 2 items, and anything > 3 would be a massive UX issue, so I'm not really concerned about this